ReportSimulationOutput/measure.rb

```diff
@@ -961,69 +961,69 @@ def get_outputs(runner, args)
     @resilience.each do |key, resilience|
       next unless key == RT::Battery
       next unless (args[:include_annual_resilience] || args[:include_timeseries_resilience])
+      next if resilience.variables.empty?
 
-      resilience.variables.map { |v| v[0] }.uniq.each do |sys_id|
-        keys = resilience.variables.select { |v| v[0] == sys_id }.map { |v| v[1] }
-        vars = resilience.variables.select { |v| v[0] == sys_id }.map { |v| v[2] }
-
-        minimum_storage_state_of_charge_fraction = nil
-        batt_kwh = nil
-        batt_kw = nil
-        batt_roundtrip_eff = nil
-        batt_loss = nil
+      resilience_frequency = 'timestep'
+      ts_per_hr = @model.getTimestep.numberOfTimestepsPerHour
+      if args[:timeseries_frequency] != 'timestep'
+        resilience_frequency = 'hourly'
+        ts_per_hr = 1
+      end
 
-        @hpxml.batteries.each do |battery|
-          @model.getElectricLoadCenterDistributions.each do |elcd|
-            battery_id = elcd.additionalProperties.getFeatureAsString('HPXML_ID')
-            next unless (battery_id.is_initialized && battery_id.get == battery.id)
+      vars = ['Electric Storage Charge Fraction']
+      keys = resilience.variables.select { |v| v[2] == vars[0] }.map { |v| v[1] }
+      batt_soc = get_report_variable_data_timeseries(keys, vars, 1, 0, resilience_frequency)
 
-            minimum_storage_state_of_charge_fraction = elcd.minimumStorageStateofChargeFraction
-          end
+      keys = ['EMS']
+      vars = resilience.variables.select { |v| v[1] == keys[0] }.map { |v| v[2] }
+      batt_loss = get_report_variable_data_timeseries(keys, vars, UnitConversions.convert(-1.0, 'J', 'kWh'), 0, resilience_frequency)
```

Comment:
This is the other part of the bugfix. Previously if you requested hourly output and we obtained a `batt_loss`, it would be a negative number. Now it's a positive number.